pkg/controller/vsphere/reconciler.go

```diff
@@ -442,6 +442,17 @@ func clone(s *machineScope) (string, error) {
 
 	resourcepool, err := s.GetSession().Finder.ResourcePoolOrDefault(s, resourcepoolPath)
 	if err != nil {
+		// TODO: move error checks to provider spec validation
+		var multipleFoundError *find.MultipleFoundError
+		var notFoundError *find.NotFoundError
+		if errors.As(err, &multipleFoundError) {
+			return "", machinecontroller.InvalidMachineConfiguration("multiple resource pools found, specify one in config")
+		}
+
+		if errors.As(err, &notFoundError) {
+			return "", machinecontroller.InvalidMachineConfiguration("resource pool not found, specify valid value")
```

Comment:
can we unit test this errors now?
we can move this to its own function if we need to.

Reply:
wouldn't this be an issue for https://github.com/openshift/machine-api-operator/pull/585/files#diff-11b6610f82e5d47fdfb8dda44b90dd1eR435-R440
as well?

Reply:
Creating and deleting resource pools in the mocked environment doesn't look easy. I'll add tests in a follow-up PR.